instance/management/commands/instance_statistics_csv.py

```diff
@@ -0,0 +1,221 @@
+            help='Path to the output file of the new CSV. Leave blank to use stdout.'
+        )
+
+    def handle(self, *args, **options):
+        # Determine the stream to be used for outputting the CSV.
+        if options['out'] is None:
+            out = self.stdout
+        else:
+            try:
+                out = open(options['out'], 'w')
+            except PermissionError:
+                self.stderr.write(self.style.ERROR(
+                    'Permission denied while attempting to write file: {outfile}'.format(outfile=options['out'])
+                ))
+                sys.exit(1)
+
+        self.collect_instance_statistics(out, options['qualified_domain'], options['days'])
+
+    def get_instance_from_qualified_domain(self, qualified_domain):
+        try:
+            instance = OpenEdXInstance.objects.get(
+                Q(external_lms_domain=qualified_domain) | Q(internal_lms_domain=qualified_domain)
+            )
+        except OpenEdXInstance.DoesNotExist:
+            self.stderr.write(self.style.ERROR(
+                'No OpenEdXInstance exists with an external or internal domain of {qualified_domain}'.format(qualified_domain=qualified_domain)
+            ))
+            sys.exit(1)
+
+        # If there are no appservers for the instance, we should error out
+        if instance.appserver_set.first() is None:
+            self.stderr.write(self.style.ERROR(
+                'No OpenEdXAppServers have been created for the instance with external or internal domain of {qualified_domain}'.format(qualified_domain=qualified_domain)
+            ))
+            sys.exit(1)
+
+        return instance
+
+    def get_elasticsearch_statistics(playbook_output_dir, name_prefix, num_days):
+        inventory = '[apps]\n{server}'.format(server=self.LOGS_SERVER)
+        playbook_path = os.path.join(settings.SITE_ROOT, 'playbooks/collect_instance_statistics/collect_elasticsearch_data.yml')
+
+        def log_line(line):
+            """Helper to pass to capture_playbook_output()."""
+            self.stderr.write(self.style.SUCCESS(line))
+        log_line.info = log_line
+        log_line.error = log_line
+
+        # Launch the collect_elasticsearch_data playbook, which places a file into the `playbook_output_dir`
+        # on this host.
+        ansible.capture_playbook_output(
+            requirements_path=os.path.join(os.path.dirname(playbook_path), 'requirements.txt'),
+            inventory_str=inventory,
+            vars_str=(
+                'local_output_dir: {output_dir}\n'
+                'remote_output_filename: /tmp/activity_report\n'
+                'server_name_prefix: {server_name_prefix}\n'
+                'num_days: {num_days}'
+            ).format(
+                output_dir=playbook_output_dir,
+                server_name_prefix=name_prefix,
+                num_days=num_days
+            ),
+            playbook_path=playbook_path,
+            username=settings.OPENSTACK_SANDBOX_SSH_USERNAME,
```

Comment:
@pcockwell, we should create a separate variable for connecting to the `LOGS_SERVER` server and use it instead.

Reply:
Now using `OPENSTACK_LOGS_SERVER_SSH_USERNAME`

Reply:
@pcockwell, the logs server has nothing to do with OpenStack. So can we get rid of the prefix?